scripts/download.py

```diff
@@ -100,5 +97,31 @@ def download_from_hub(
         convert_hf_checkpoint(checkpoint_dir=directory, dtype=dtype)
 
 
+def find_weight_files(repo_id: str, access_token: Optional[str]) -> Tuple[List[str], List[str]]:
+    from huggingface_hub import repo_info
+    from huggingface_hub.utils import filter_repo_objects
+
+    with gated_repo_catcher(repo_id):
+        info = repo_info(repo_id, token=access_token)
+    filenames = [f.rfilename for f in info.siblings]
+    bins = list(filter_repo_objects(items=filenames, allow_patterns=["*.bin*"]))
+    safetensors = list(filter_repo_objects(items=filenames, allow_patterns=["*.safetensors"]))
+    return bins, safetensors
```

Comment:
Is it necessary to return lists, since you only need to know whether the weights are in `.bin` or in `.safetensors` format?

```python
from huggingface_hub import repo_info

with gated_repo_catcher(repo_id):
        info = repo_info(repo_id, token=access_token)
return any(f.rfilename.endswith(".safetensors") for f in info.siblings)
```

or even move the logic out of this function, since it's used in the single place only.

Reply:
Not that it matters, but the current format lets us raise an error if there were no binaries and no safetensors

Reply:
Good point.